src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -0,0 +1,98 @@
+            const auto& parent_expr = parent_port.get_expr();
+            const auto parent_loop_ids = parent_expr->get_loop_ids();
+            if (parent_loop_ids.empty())
+                continue;
+
+            const auto& parent_loop_id = parent_loop_ids.front();
+            const auto parent_loop_port = loop_manager->get_loop_port_by_expr_port(parent_port, parent_loop_id);
+            // We don't split loop which are not compatible with parent loop because such loops will not be fused
+            if (parent_loop_port.is_incremented != entry_point.is_incremented)
+                continue;
+
+            const auto parent_loop = loop_manager->get_loop_info(parent_loop_id);
+            if (can_be_split(loop, parent_loop)) {
+                loop_was_split = true;
+                const bool split_parent = parent_loop->increment < loop->increment;
+                const auto& loop_to_split = split_parent ? parent_loop : loop;
+                const auto& loop_to_split_id = split_parent ? parent_loop_id : loop_id;
+                const auto& loop_to_fuse = !split_parent ? parent_loop : loop;
+
+                loop_to_split->work_amount = loop_to_fuse->increment;
+                const auto split_loop = std::make_shared<LoopManager::LoopInfo>(loop_to_fuse->work_amount,
+                                                                                loop_to_fuse->increment,
+                                                                                loop_to_split->dim_idx,
+                                                                                loop_to_split->entry_points,
+                                                                                loop_to_split->exit_points);
+                split_loop->outer_splited_loop = true;
+                LinearIR::constExprIt loop_begin_pos, loop_end_pos;
+                LoopManager::get_loop_bounds(linear_ir,
+                                             loop_to_split->entry_points,
+                                             loop_to_split->exit_points,
+                                             loop_begin_pos,
+                                             loop_end_pos,
+                                             loop_to_split_id);
+
+                const auto new_loop_id = loop_manager->add_loop_info(split_loop);
+                for (auto it = loop_begin_pos; it != loop_end_pos; it++) {
+                    const auto& iexpr = *it;
+                    // Note: There could be exprs inside loop bounds that don't belong to the loop
+                    if (iexpr->get_loop_ids().front() == loop_to_split_id) {
+                        auto split_loop_ids = iexpr->get_loop_ids();
+                        split_loop_ids.insert(split_loop_ids.begin(), new_loop_id);
+                        iexpr->set_loop_ids(split_loop_ids);
+                    }
+                }
+                break;
```

Comment:
Can we use `mark_loop` here? We really should avoid the situations when the user has to call `set_loop_ids` himself. Only `LoopManager` should be worried about how to mark loops

Reply:
`mark_loop` has been reused, thanks. Please double check